packages/extract/src/MalabiExtract.ts

```diff
@@ -0,0 +1,137 @@
+import { ReadableSpan } from '@opentelemetry/tracing';
+import { SpanKind } from '@opentelemetry/api';
+import { SemanticAttributes, MessagingOperationValues } from '@opentelemetry/semantic-conventions';
+import { MalabiSpan } from './MalabiSpan';
+
+class MalabiExtract {
+    private spans: ReadableSpan[];
+
+    constructor(spans: ReadableSpan[]) {
+        this.spans = spans;
+    }
+
+    private filter(predicate: (span: ReadableSpan) => boolean): MalabiExtract {
+        return new MalabiExtract(this.spans.filter(predicate));
+    }
+
+    get raw() {
+        return this.spans;
+    }
+
+    get length() {
+        return this.spans.length;
+    }
+
+    get first() {
+        if (this.spans.length < 1) throw new Error(`Tried to get the "first" span, but there are no spans.`);
+        return new MalabiSpan(this.spans[0]);
+    }
+
+    get second() {
+        if (this.spans.length < 2)
+            throw new Error(`Tried to get the "second" span, bt there are only ${this.spans.length} spans.`);
+        return new MalabiSpan(this.spans[1]);
+    }
+
+    at(index: number) {
+        if (this.spans.length < index + 1)
+            throw new Error(`Tried to get the span in ${index} index, but there are only ${this.spans.length} spans.`);
+        return new MalabiSpan(this.spans[index]);
+    }
+
+    http() {
+        return new MalabiExtract(this.spans.filter((span) => Boolean(span.attributes[SemanticAttributes.HTTP_METHOD])));
+    }
+
+    httpMethod(method: string) {
+        return this.filter(
+            (span) =>
+                (span.attributes[SemanticAttributes.HTTP_METHOD] as string)?.toLowerCase() === method.toLowerCase()
+        );
+    }
+
+    httpGet() {
+        return this.filter(
+            (span) => (span.attributes[SemanticAttributes.HTTP_METHOD] as string)?.toLowerCase() === 'get'
+        );
+    }
+
+    httpPost() {
+        return this.filter((span) => (span.attributes[SemanticAttributes.HTTP_METHOD] as string)?.toLowerCase() === 'post');
+    }
+
+    route(r: string) {
+        return this.filter((span) => (span.attributes[SemanticAttributes.HTTP_ROUTE] as string)?.toLowerCase() === r.toLowerCase());
+    }
+
+    path(p: string) {
+        return this.filter((span) => (span.attributes['http.path'] as string)?.toLowerCase() === p.toLowerCase());
+    }
+
+    messagingSend() {
+        return this.filter(span => span.kind === SpanKind.PRODUCER);
+    }
+
+    messagingReceive() {
+        return this.filter(span => span.attributes[SemanticAttributes.MESSAGING_OPERATION] === MessagingOperationValues.RECEIVE)
+    }
+
+    messagingProcess() {
+        return this.filter(span => span.attributes[SemanticAttributes.MESSAGING_OPERATION] === MessagingOperationValues.PROCESS)
+    }
+
+    sqs() {
```

Comment:
I would add 'aws' as a prefix.
It's easy for us to understand but for regular user it might be good to add more context